src/utils/pgwire/src/pg_response.rs

```diff
@@ -124,7 +127,12 @@ impl StatementType {
     pub fn is_dml(&self) -> bool {
```

Comment:
There's another function named `flush_for_write`. How about using `is_dml` for that?

Reply:
> Could you please also add some end-to-end tests?

Have added some. Those tests would pass should[this change to sqllogictest](https://github.com/risinglightdb/sqllogictest-rs/pull/141) be merged.